compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -828,18 +829,32 @@ checkDefTypeSyn DefTypeSyn{synParams,synType} = do
   where
     base = checkType synType KStar
 
+
 -- | Check that an interface definition is well defined.
 checkIface :: MonadGamma m => Module -> DefInterface -> m ()
-checkIface m DefInterface{..} = do
-  checkUnique (EDuplicateInterfaceChoiceName intName) $ NM.names intFixedChoices
-  checkUnique (EDuplicateInterfaceMethodName intName) $ NM.names intMethods
-  forM_ intRequires (inWorld . lookupInterface) -- verify that required interface exists
-  forM_ intMethods checkIfaceMethod
+checkIface m iface = do
+  let tcon = Qualified PRSelf (moduleName m) (intName iface)
+
+  -- check requires
+  when (tcon `S.member` intRequires iface) $
+    throwWithContext (ECircularInterfaceRequires (intName iface) Nothing)
+  forM_ (intRequires iface) $ \requiredIfaceId -> do
+    requiredIface <- inWorld (lookupInterface requiredIfaceId)
+    when (tcon `S.member` intRequires requiredIface) $
+      throwWithContext (ECircularInterfaceRequires (intName iface) (Just requiredIfaceId))
```

Comment:
Not a big issue, but we actually does not need this case.
If you require the requires is transitively closed, you just need to check that an interface does not require itself (done at line 840).

Reply:
Yep, that's why I said it's only done to get a better error message.

Reply:
I agree. I think it's better for the check to be simpler in the interpreter. The compiler is more user-facing.